### PR TITLE
SW-4488  Projects List View and Empty state

### PR DESCRIPTION
**Pull request by constanzauanini**
Adding the table with projects and the empty state when no projects.

### CODE REVIEW

File: src/components/Projects/index.tsx [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+  const history = useHistory();
+  const [temporalSearchValue, setTemporalSearchValue] = useState('');
+  const debouncedSearchTerm = useDebounce(temporalSearchValue, 250);
+  const [results, setResults] = useState<Project[]>();
+  const { isMobile } = useDeviceInfo();
+  const contentRef = useRef(null);
+  const { activeLocale } = useLocalization();
+
+  const search = useCallback(
+    async (searchTerm: string, skipTfContact = false) => {
```
Comment (karthikbtf) [inline src/components/Projects/index.tsx:63]: can remove the skipTfContact

File: src/components/Projects/index.tsx [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+          },
+        ],
+        count: 0,
+      };
+
+      if (searchField) {
+        params.search.children.push(searchField);
+      }
+
+      const searchResults = await SearchService.search(params);
```
Comment (karthikbtf) [inline src/components/Projects/index.tsx:100]: Do we want to use redux here, I understand it's a copy of existing code (which isn't the best to use as a template)... 
either move all this into projects service and just call the function with search term and org id, which returns back a list of Project[]?
Or use redux..?

File: src/services/ProjectsService.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  );
+
+  return response;
+};
+
+/**
+ * Exported functions
+ */
+const ProjectsService = {
+  listProjects,
```
Comment (karthikbtf) [inline src/services/ProjectsService.ts:45]: to be used later?

**Review by karthikbtf — COMMENTED** [review]
Mostly good, some thoughts.

File: src/components/Projects/index.tsx (hunk 2 above)
Comment (nickgraz) [reply src/components/Projects/index.tsx:100]: I agree it should be moved. I think in the past I've been putting these sorts of search queries into a service and using redux more for 1:1 entity APIs. Is that correct? What is the general pattern to deciding whether it goes into a service and is used in a component's state versus redux state?

Comment (karthikbtf) [reply src/components/Projects/index.tsx:100]: Redux was recently introduced into the codebase with the idea that all new functionality would 'ideally' use redux.
We didn't consider differentiating between using component state vs redux state for business data - maybe we should be open to that for sure.
As far as what goes into a service, anything involving wrangling types and parsing responses into types, etc. Search is a very good use-case, not sure components need to be aware search request payload semantics.. good candidate for services?

Comment (karthikbtf) [reply src/components/Projects/index.tsx:100]: @constanzauanini , at a minimum move the 'payload creation', search call and 'response parsing' into a service function. Should be good to go after that.

Comment (constanzauanini) [reply src/components/Projects/index.tsx:100]: Yes, I just copied existing code that was part of a component. But it should be part of a service. Will move it

File: src/services/ProjectsService.ts (hunk 3 above)
Comment (constanzauanini) [reply src/services/ProjectsService.ts:45]: I'm using it to show the empty state if no projects

File: src/services/ProjectsService.ts [hunk 4]
```diff
@@ -38,11 +40,55 @@ const listProjects = async (organizationId: number, locale?: string | null): Pro
   return response;
 };
 
+/**
+ * Search projects
+ */
+const searchProjects = async (organizationId: number, query?: string): Promise<SearchResponseElement[] | null> => {
```
Comment (karthikbtf) [inline src/services/ProjectsService.ts:46]: can you return Project[] | null

File: src/components/Projects/index.tsx [hunk 5]
```diff
@@ -60,54 +59,9 @@ export default function ProjectsList(): JSX.Element {
...
-          name: result.name as string,
-          description: result.description as string,
-          id: result.id as number,
-          organizationId: result.organizationId as number,
-        });
-      });
-      return projectResults;
+    async (searchTerm: string) => {
+      const searchResults = await ProjectsService.searchProjects(selectedOrganization.id, searchTerm);
+      return searchResults as Project[];
```
Comment (karthikbtf) [inline src/components/Projects/index.tsx:64]: do the casting as Project[] in the service call instead

**Review by karthikbtf — APPROVED** [review]
lgtm, one suggestion and feel free to merge after that

**Review by karthikbtf — APPROVED** [review]
🎉

**Pull request merged**